src/components/Common/AvatarEditModal.tsx

```diff
@@ -78,51 +81,73 @@ const AvatarEditModal = ({
     VideoConstraints.user,
   );
   const { t } = useTranslation();
+  const { captureWebcamImage, processCroppedImage } = useImageCapture();
   const [isDragging, setIsDragging] = useState(false);
   const { requestPermission } = useMediaDevicePermission();
+  const [isCropping, setIsCropping] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
   const handleSwitchCamera = useCallback(() => {
     setConstraint(
       constraint.facingMode === "user"
         ? VideoConstraints.environment
         : VideoConstraints.user,
     );
-  }, []);
+  }, [constraint]);
```

Comment:
This dependency shouldn't be necessary here since function isn't changing when constraint changes.